odrive_state_controller/src/odrive_state_controller.cpp

```diff
@@ -60,6 +62,17 @@ void OdriveStateController::starting(const ros::Time& time) {
 }
 
 void OdriveStateController::update(const ros::Time& time, const ros::Duration& /*period*/) {
+  static bool clear_errors_last = false;
+
+  const bool clear_errors_current = !do_not_clear_errors_flag_.test_and_set();
+
+  // Only trigger clear_errors once per service call
+  const bool clear_errors = clear_errors_current && !clear_errors_last;
```

Comment:
since you're `test_and_set()`ing, I think this rising edge detection is unnecessary.

Reply:
this is true

Reply:
fixed